src/library/scala/collection/Iterator.scala

```diff
@@ -32,12 +32,14 @@ object Iterator {
     def traversableToColl[B](t: GenTraversable[B]) = t.toIterator
   }
 
-  /** The iterator which produces no values. */
-  val empty: Iterator[Nothing] = new AbstractIterator[Nothing] {
+  private[this] final val _empty: Iterator[Nothing] = new AbstractIterator[Nothing] {
     def hasNext: Boolean = false
     def next(): Nothing = throw new NoSuchElementException("next on empty iterator")
   }
 
+  /** The iterator which produces no values. */
```

Comment:
@retronym we need to ensure this is inlined as a getter. Similar issue to "tl" in :: ?

Reply:
These no-runtime-op no-need-to-override methods are all marked as `@inline final` in the strawman. The same should be done for this method.

Reply:
Thanks for suggestion, method now marked as `@inline final`.

Reply:
That is the solution to a different problem unfortunately
see https://github.com/rorygraves/scalac_perf/issues/46
so this should probably be 
`@inline final def empty[T]: Iterator[T] = _empty: @inline`

Reply:
Thanks a lot for suggestion but it looks like current implementation also gets inlined. This is what I see in the bytecode of client code that uses it:

```
public final class scala.collection.IteratorInlineTest {
  public static scala.collection.Iterator<java.lang.Object> inlineTest2$default$1();
    Code:
       0: getstatic     #16                 // Field scala/collection/IteratorInlineTest$.MODULE$:Lscala/collection/IteratorInlineTest$;
       3: invokevirtual #18                 // Method scala/collection/IteratorInlineTest$.inlineTest2$default$1:()Lscala/collection/Iterator;
       6: areturn

  public static scala.collection.Iterator<java.lang.Object> inlineTest3();
    Code:
       0: getstatic     #16                 // Field scala/collection/IteratorInlineTest$.MODULE$:Lscala/collection/IteratorInlineTest$;
       3: invokevirtual #21                 // Method scala/collection/IteratorInlineTest$.inlineTest3:()Lscala/collection/Iterator;
       6: areturn

  public static scala.collection.Iterator<java.lang.Object> inlineTest2(scala.collection.Iterator<java.lang.Object>);
    Code:
       0: getstatic     #16                 // Field scala/collection/IteratorInlineTest$.MODULE$:Lscala/collection/IteratorInlineTest$;
       3: aload_0
       4: invokevirtual #26                 // Method scala/collection/IteratorInlineTest$.inlineTest2:(Lscala/collection/Iterator;)Lscala/collection/Iterator;
       7: areturn

  public static scala.collection.immutable.List<java.lang.Object> inlineTest1();
    Code:
       0: getstatic     #16                 // Field scala/collection/IteratorInlineTest$.MODULE$:Lscala/collection/IteratorInlineTest$;
       3: invokevirtual #30                 // Method scala/collection/IteratorInlineTest$.inlineTest1:()Lscala/collection/immutable/List;
       6: areturn
}
```

As we can see we __get static field value__ every time (no __invokestatic__) so it looks fine.